tools/run_tests/helper_scripts/build_cxx.bat

```diff
@@ -45,7 +45,7 @@ If "%GRPC_CMAKE_GENERATOR%" == "Ninja" (
 ) else (
   @rem Use one of the Visual Studio generators.
 
-  cmake -G "%GRPC_CMAKE_GENERATOR%" -A "%GRPC_CMAKE_ARCHITECTURE%" -DgRPC_BUILD_TESTS=ON -DgRPC_BUILD_MSVC_MP_COUNT=%GRPC_RUN_TESTS_JOBS% %* ../.. || goto :error
+  cmake -G "%GRPC_CMAKE_GENERATOR%" -A "%GRPC_CMAKE_ARCHITECTURE%" -DCMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE=x64 -DgRPC_BUILD_TESTS=ON -DgRPC_BUILD_MSVC_MP_COUNT=%GRPC_RUN_TESTS_JOBS% %* ../.. || goto :error
```

Comment:
This might work, but it's still unclear to me whether the build failure we're seeing only happens when building using visual studio generator (which this change could fix) and it doesn't happen in ninja builds (which are unaffected by this change).

In fact, I think a saw ninja builds that suffer from the same problem and therefore should be also fixed:
https://source.cloud.google.com/results/invocations/0d427a67-8e2b-4374-b04a-43b21e43b553/targets;collapsed=/github%2Fgrpc%2Ftoplevel_run_tests_invocations%2Frun_tests_c_windows_dbg_native/tests

Reply:
also note that CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE only exists starting from cmake 3.8 
https://cmake.org/cmake/help/latest/variable/CMAKE_VS_PLATFORM_TOOLSET_HOST_ARCHITECTURE.html, but currently the minimum cmake version in the CMakeLists.txt is lower (I think 3.5)

Reply:
As my comment I left below, it turned out that all builds use Ninja so I need to fix for Ninja actually.